common/common.go

```diff
@@ -7,3 +7,30 @@ import (
 func ServiceTag(sv string) tag.ZapTag {
 	return tag.NewStringTag("service", sv)
 }
+
+// GCD calulates the greatest common divisor using the Euclidean algorithm.
+//
+// https://en.wikipedia.org/wiki/Euclidean_algorithm#Implementations
+func GCD(a, b int32) int32 {
+	if a == 0 || b == 0 {
+		return 0
+	}
+	if a > b {
+		a, b = b, a
+	}
+
+	for b != 0 {
+		a, b = b, a%b
+	}
+	return a
+}
+
+// LCM calcuates the least common multiple of a and b.
+//
+// https://en.wikipedia.org/wiki/Least_common_multiple#Using_the_greatest_common_divisor
+func LCM(a, b int32) int32 {
```

Comment:
There are surprisingly few small libraries for doing this, and I wasn't satisfied with their unit testing (just few cases). I figured these functions are so small that I'd implement them myself to avoid the dependency.